cogito/protocol.go

```diff
@@ -199,13 +200,24 @@ const (
 
 const KeyState = "state"
 
-// Validate checks whether the build state, parsed from JSON, is valid.
-func (bs BuildState) Validate() error {
-	switch bs {
+func (bs *BuildState) UnmarshalJSON(data []byte) error {
```

Comment:
To better understand what is going on here, see for example https://pkg.go.dev/encoding/json@go1.19#example-package-CustomMarshalJSON